tools/clang/test/CodeGenSPIRV/bezier.hull.hlsl2spv

```diff
@@ -191,6 +191,8 @@ BEZIER_CONTROL_POINT SubDToBezierHS(InputPatch<VS_CONTROL_POINT_OUTPUT, MAX_POIN
 // %int_3 = OpConstant %int 3
 // %float_5 = OpConstant %float 5
 // %float_6 = OpConstant %float 6
+// %121 = OpConstantNull %HS_CONSTANT_DATA_OUTPUT
```

Comment:
As discussed offline, I don't understand why these are generated... The shader in this case has 'return' statements and therefore, `isCurrentBasicBlockTerminated()` should be true for all BBs, and no Null returns should occur...

Reply:
It turns out to be the side-effect of how we handle `ReturnStmt`... We always create a new basic block for potentially more stuff after translating a `ReturnStmt`: https://github.com/Microsoft/DirectXShaderCompiler/pull/809/commits/d745f47b6e84233d713bfd5696ce2e57536200e8#diff-e115cf7094756003c1fba8470b9d650aL1175

I've added an ad-hoc check to avoid doing that at the end of the function. I think it'll cover the majority of the cases.